test-poc/framework/src/main/java/org/keycloak/test/framework/server/DefaultKeycloakTestServerConfig.java

```diff
@@ -1,5 +1,17 @@
 package org.keycloak.test.framework.server;
 
+import java.util.Optional;
+
 public class DefaultKeycloakTestServerConfig implements KeycloakTestServerConfig {
 
+    @Override
```

Comment:
Not sure you'd ever want to configure the admin username and password here, but rater it should be external config.

This change doesn't really seem all that relevant to injecting a webdriver though, so complicating the review/discussions around this PR.

Reply:
It's needed for https://github.com/keycloak/keycloak/pull/30378#discussion_r1647412929, it's relevant for the testing of the welcome page. If you like I can change it so `KeycloakTestServerConfig` has defaults for this, and then override these methods in `NoAdminUserKeycloakTestServerConfig` and return an empty optional.